src/common/tagging/external/tag_group.go

```diff
@@ -0,0 +1,209 @@
+package external
+
+import (
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"strconv"
+
+	"github.com/bridgecrewio/yor/src/common/logger"
+	"github.com/bridgecrewio/yor/src/common/structure"
+	"github.com/bridgecrewio/yor/src/common/tagging"
+	"github.com/bridgecrewio/yor/src/common/tagging/tags"
+	"github.com/bridgecrewio/yor/src/common/utils"
+	"gopkg.in/yaml.v2"
+)
+
+type TagGroup struct {
+	tagging.TagGroup
+	configFilePath  string
+	config          map[interface{}]interface{}
+	tagGroupsByName map[string][]Tag
+}
+
+type Tag struct {
+	tags.ITag
+	defaultValue string
+	filters      map[interface{}]interface{}
+	matches      []interface{}
+}
+
+func (t Tag) SatisfyFilters(block structure.IBlock, tagFilterDir string) bool {
```

Comment:
@tronxd seems like a method we should look towards moving to the base tag package. not for this PR